scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt

```diff
@@ -383,9 +385,36 @@ internal class ScenarioRunServiceImpl(
     val workspace = workspaceService.findWorkspaceById(organizationId, workspaceId)
     sendScenarioRunMetaData(organizationId, workspace, scenarioId, scenarioRun.csmSimulationRun)
 
+    if (startInfo.runTemplate.deleteLastRunData == true) {
+      // Start a thread to check whether a scenariorun is completed
+      // then delete the previous simulation results
+      logger.debug("Start coroutine to poll simulation status")
+      GlobalScope.launch {
+        withTimeout(CSM_ARGO_WORKFLOWS_TIMEOUT.toLong()) {
+          deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(scenarioRun)
+        }
+      }
+      logger.debug("Coroutine to poll simulation status launched")
+    }
     return scenarioRun.withoutSensitiveData()!!
   }
 
+  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
```

Comment:
Please do not write `while(true)` it's awful.
try something like this
```
  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
    val scenarioRunId = currentRun.id!!
    val workspaceId = currentRun.workspaceId!!
    val organizationId = currentRun.organizationId!!
    val scenarioId = currentRun.scenarioId!!
    val csmSimulationRun = currentRun.csmSimulationRun

    var scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    while (
      scenarioRunStatus.state != ScenarioRunState.Successful ||
      scenarioRunStatus.state != ScenarioRunState.Failed ||
      scenarioRunStatus.state != ScenarioRunState.DataIngestionFailure ) {
      logger.info("ScenarioRun {} is still running, waiting for purging data", csmSimulationRun)
      Thread.sleep(10000)
      scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    }

    if (scenarioRunStatus.state == ScenarioRunState.Successful) {
      logger.info("ScenarioRun {} is Successfull => purging data", csmSimulationRun)
      deleteScenarioRunsByScenario(organizationId, workspaceId, scenarioId)
    } else {
      logger.info("ScenarioRun {} is in error {} => no purging data",
        csmSimulationRun,
        scenarioRunStatus.state)
    }

  }
  ```
  Note that the value 10000 should be configurable

Reply:
It's a sore in the eye
with the following code the intention is clear and neat

 private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
    while (true) {
      Thread.sleep(10000)
      val scenarioRunStatus = getScenarioRunStatus(currentRun.organizationId!!, currentRun.id!!)

      if (scenarioRunStatus.state == ScenarioRunState.Successful) {
        deleteScenarioRunsByScenario(
            currentRun.organizationId!!, currentRun.workspaceId!!, currentRun.scenarioId!!)
        return
      } else if (scenarioRunStatus.state == ScenarioRunState.DataIngestionFailure ||
          scenarioRunStatus.state == ScenarioRunState.Failed) {
        logger.info("ScenarioRun {} failed", currentRun.csmSimulationRun)
        break
      }
    }
  }